### PR TITLE
Fix(ticket#227) :Added Headers in Members Pages

### PR DESCRIPTION
**Pull request by Mubashar1009**
## Change Request

Add Headers in Edit and Detail Member page 

### Implementation

* Updated `MemberHeader` Component and use it for all member routes 
* Fixed loading issue in Member List 
* Fixed Member List Updating issue after deleting a member 
* Added Date Format in Detail Member page 
### Resources
image
image
image

### CODE REVIEW

File: src/app/(app)/members/components/MemberList.tsx [hunk 1]
```diff
@@ -59,7 +58,7 @@ export const MemberList = () => {
       accessorKey: "role",
       header: "Role",
       cell: ({ row }) => (
-        <div className="capitalize">{row.getValue("role")}</div>
+        <div className="capitalize">{formatRole(row.getValue("role"))}</div>
```
Comment (nadirhussain786) [inline src/app/(app)/members/components/MemberList.tsx:61]: @Mubashar1009  why we need to use `formatRole`?

File: src/app/(app)/members/components/MemberFormBuilder.tsx [hunk 2]
```diff
@@ -312,6 +324,7 @@ export const MemberFormBuilder = ({ member }: MemberFormBuilder) => {
                       <Button
                         data-testid="ehad_duration"
                         variant={"outline"}
+                        onClick={()=>{dateformatter(field.value)}}
                         className={cn(
                           `justify-start text-left font-normal text-xs
                           ${field.value} && "text-muted-foreground
```
Comment (nadirhussain786) [inline src/app/(app)/members/components/MemberFormBuilder.tsx:330]: @Mubashar1009  This condition is not looking good.

File: src/app/(app)/members/components/DetailMemberHeader.tsx [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+  const pathname = usePathname();
+  const router = useRouter();
+
+  const [isSidebarOpen, setSidebarOpen] = useState<boolean>(false);
+
+  const toggleSidebar = () => {
+    setSidebarOpen(!isSidebarOpen);
+  };
+
+  useEffect(() => {
```
Comment (muhammad-bilal-ali-khokhar) [inline src/app/(app)/members/components/DetailMemberHeader.tsx:21]: no need of this `useEffect` you can handle through breadcrums

File: src/app/(app)/members/components/EditMemberBreadCrumbs.tsx
Comment (muhammad-bilal-ali-khokhar) [inline src/app/(app)/members/components/EditMemberBreadCrumbs.tsx:1]: No need of seprate breadcum

File: src/app/(app)/members/components/EditMemberHeader.tsx [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+import React, { useEffect, useState } from "react";
+import { Sidebar } from "@/app/layout/components/Sidebar";
+import { Button } from "@/components/ui/button";
+import { HamburgerMenuIcon } from "@radix-ui/react-icons";
+import { usePathname, useRouter } from "next/navigation";
+import { Routes } from "@/lib/routes";
+import { EditMemberBreadCrumbs } from "./EditMemberBreadCrumbs";
+
+export const  EditMemberHeader = ({ id }: { id: string }) => {
```
Comment (muhammad-bilal-ali-khokhar) [inline src/app/(app)/members/components/EditMemberHeader.tsx:11]: use here `useParms`

File: src/app/(app)/members/components/EditMemberHeader.tsx [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+  const pathname = usePathname();
+  const router = useRouter();
+
+  const [isSidebarOpen, setSidebarOpen] = useState<boolean>(false);
+
+  const toggleSidebar = () => {
+    setSidebarOpen(!isSidebarOpen);
+  };
+
+  useEffect(() => {
```
Comment (muhammad-bilal-ali-khokhar) [inline src/app/(app)/members/components/EditMemberHeader.tsx:21]: i mentioned above, no need of this

File: src/app/(app)/members/components/MemberDetails.tsx [hunk 6]
```diff
@@ -12,8 +12,10 @@ import {
 import { UserDetails } from "@/types";
 import { MemberRole } from "@/constant/constant";
 import { getAge } from "@/utils/ageFormater";
+import { dateformatter } from "@/utils/dateFormatter";
```
Comment (muhammad-bilal-ali-khokhar) [inline src/app/(app)/members/components/MemberDetails.tsx:15]: which date formatter? Please chose a suitable variable name

File: src/app/(app)/members/components/MemberDetails.tsx [hunk 7]
```diff
@@ -92,15 +94,15 @@ export const MemberDetails = ({ userInfo }: { userInfo: UserDetails }) => {
               <p className="text-xs">
                 <span className="font-medium text-muted-foreground">
                   Ehad Duration:
-                </span>{" "}
-                {userInfo.ehad_duration} years
+                </span>
+                { dateformatter(new Date(userInfo.ehad_duration))  }
```
Comment (muhammad-bilal-ali-khokhar) [inline src/app/(app)/members/components/MemberDetails.tsx:98]: define a varable at top and use here ,

File: src/app/(app)/members/components/MemberFormBuilder.tsx [hunk 8]
```diff
@@ -136,11 +138,20 @@ export const MemberFormBuilder = ({ member }: MemberFormBuilder) => {
...
+
+
+
+
+
   const {
     formState: { errors },
   } = form;
 
+  const dateformatter = (date: Date) => {
```
Comment (muhammad-bilal-ali-khokhar) [inline src/app/(app)/members/components/MemberFormBuilder.tsx:150]: I'm confused , why you using this function , even though you are returing value as it is.

File: src/app/(app)/members/components/MemberFormBuilder.tsx [hunk 9]
```diff
@@ -312,6 +324,7 @@ export const MemberFormBuilder = ({ member }: MemberFormBuilder) => {
                       <Button
                         data-testid="ehad_duration"
                         variant={"outline"}
+                        onClick={()=>{dateformatter(field.value)}}
```
Comment (muhammad-bilal-ali-khokhar) [inline src/app/(app)/members/components/MemberFormBuilder.tsx:327]: I'll appricate you if you keep this is variable

File: src/app/(app)/members/edit/[id]/page.tsx [hunk 10]
```diff
@@ -9,8 +10,15 @@ type Parameters = {
 };
 const Page = async ({ params }: Parameters) => {
   const { id } = params;
+
   const member = await getUserById(id);
-  return <MemberFormBuilder member={member} />;
+
+  return (
+    <>
+      <EditMemberHeader id={id}/>
```
Comment (muhammad-bilal-ali-khokhar) [inline src/app/(app)/members/edit/[id]/page.tsx:18]: why header needs `id`

File: src/app/(app)/members/components/DetailMemberBreadCrumbs.tsx [hunk 11]
```diff
@@ -0,0 +1,12 @@
+import React from "react";
+import { BasedBreadCrumb } from "@/common/BasedBreadCrumb/BasedBreadCrumb";
+import { Routes } from "@/lib/routes";
+import { Breadcrumbs } from "@/types";
+
+export const DetailMemberBreadCrumbs = ({ id }: { id: string }) => {
+  const breadcrumbs: Breadcrumbs[] = [
+    { href: Routes.Members, label: "Members" },
+    { href: `${Routes.MemberDetails}/${id}`, label: "Member Detail" },
```
[the remark this answers is not in the document]
Comment (Mubashar1009) [reply src/app/(app)/members/components/DetailMemberBreadCrumbs.tsx:9]: fixed

File: src/app/(app)/members/components/DetailMemberHeader.tsx [hunk 12]
```diff
@@ -0,0 +1,53 @@
...
+
+import React, { useEffect, useState } from "react";
+import { Sidebar } from "@/app/layout/components/Sidebar";
+import { Button } from "@/components/ui/button";
+import { HamburgerMenuIcon } from "@radix-ui/react-icons";
+import { usePathname, useRouter } from "next/navigation";
+import { Routes } from "@/lib/routes";
+import { DetailMemberBreadCrumbs } from "./DetailMemberBreadCrumbs";
+
+export const DetailMemberHeader = ({ id }: { id: string }) => {
```
[the remark this answers is not in the document]
Comment (Mubashar1009) [reply src/app/(app)/members/components/DetailMemberHeader.tsx:11]: actually i have update this component now this id is not using here, i have removed this now

File: src/app/(app)/members/components/DetailMemberHeader.tsx (hunk 3 above)
Comment (Mubashar1009) [reply src/app/(app)/members/components/DetailMemberHeader.tsx:21]: fixe

File: src/app/(app)/members/components/EditMemberBreadCrumbs.tsx
Comment (Mubashar1009) [reply src/app/(app)/members/components/EditMemberBreadCrumbs.tsx:1]: fixed

File: src/app/(app)/members/components/EditMemberHeader.tsx (hunk 5 above)
Comment (Mubashar1009) [reply src/app/(app)/members/components/EditMemberHeader.tsx:21]: fixed

File: src/app/(app)/members/components/MemberDetails.tsx (hunk 6 above)
Comment (Mubashar1009) [reply src/app/(app)/members/components/MemberDetails.tsx:15]: fixed

File: src/app/(app)/members/components/MemberDetails.tsx (hunk 7 above)
Comment (Mubashar1009) [reply src/app/(app)/members/components/MemberDetails.tsx:98]: fixed

File: src/app/(app)/members/components/MemberFormBuilder.tsx (hunk 8 above)
Comment (Mubashar1009) [reply src/app/(app)/members/components/MemberFormBuilder.tsx:150]: fixed

File: src/app/(app)/members/components/MemberFormBuilder.tsx (hunk 9 above)
Comment (Mubashar1009) [reply src/app/(app)/members/components/MemberFormBuilder.tsx:327]: fixed

File: src/app/(app)/members/edit/[id]/page.tsx (hunk 10 above)
Comment (Mubashar1009) [reply src/app/(app)/members/edit/[id]/page.tsx:18]: fixed

**Review by muhammad-bilal-ali-khokhar — APPROVED** [review]

**Pull request merged**